challenge/files/12_context.toml

```diff
@@ -0,0 +1,95 @@
+name = "Context"
+category = "easy"
+default_code_snippet = """
+var (
+	errTimeout error = errors.New("Timeout")
+)
+
+func main() {
```

Comment:
Code does not compiles because of unused variables lets omit its

```go
var (
	errTimeout error = errors.New("Timeout")
)

func main() {
	cancelCtx, cancel := context.WithCancel(context.Background())
	timeoutCtx, _ := context.WithTimeoutCause(context.Background(), 2*time.Second, errTimeout)
	deadlineCtx, _ := context.WithDeadline(context.Background(), time.Now().Add(2*time.Second))
	childCtx := context.WithValue(cancelCtx, "Key", "Value")
	time.AfterFunc(2*time.Second, func() {
		cancel()
	})
	go func() {
		<-timeoutCtx.Done()
		fmt.Println("Timeout context")
		fmt.Printf("%v\n", context.Cause(timeoutCtx))
		fmt.Printf("%v\n", timeoutCtx.Err())
	}()
	go func() {
		<-deadlineCtx.Done()
		fmt.Println("Deadline context")
	}()
	go func() {
		<-childCtx.Done()
		fmt.Println("Child context")
		fmt.Println(childCtx.Value("Key"))
	}()
	time.Sleep(3 * time.Second)
}
```